franklin/builder/models.py

```diff
@@ -184,41 +204,43 @@ class Environment(models.Model):
         max_length=3, choices=STATUS_CHOICES, default=REGISTERED)
     
     def build(self):
-        url = os.environ['BUILDER_URL']
-        headers = {'content-type': 'application/json'}
-        body = {
-                    "deploy_key": self.site.deploy_key,
-                    "branch": self.current_deploy.branch,
-                    "tag": self.current_deploy.tag,
-                    "git_hash": self.current_deploy.git_hash,
-                    "repo_owner": self.site.owner.name,
-                    "path": self.current_deploy.path,
-                    "repo_name": self.site.name
-                }
-        r = None
-        try:
-            r = requests.post(url, data=json.dumps(body), headers=headers)
-        except (ConnectionError, HTTPError, Timeout) as e:
-            logger.error('Connection exception : %s', e)
-        except:
-            logger.error('Unexpected Builder error: %s', sys.exc_info()[0])
-
-        if r is not None:
-            if (status.is_success(r.status_code) and 
-                    r.headers['Content-Type'] == 'application/json'):
-                building_status = r.json()['building']
-                if building_status:
-                    self.status = self.BUILDING
-                    self.save()
-                    return
+        if self.current_deploy:
+            is_tag_build = hasattr(self.current_deploy, 'tagbuild')
+            url = os.environ['BUILDER_URL']
+            headers = {'content-type': 'application/json'}
+            body = {
+                        "deploy_key": self.site.deploy_key,
+                        "branch": self.current_deploy.branch if not is_tag_build else '',
+                        "tag": self.current_deploy.tag if is_tag_build else '',
+                        "git_hash": self.current_deploy.git_hash  if not is_tag_build else '',
+                        "repo_owner": self.site.owner.name,
+                        "path": self.current_deploy.path,
+                        "repo_name": self.site.name
+                    }
+            r = None
```

Comment:
This is the slightly clever/complicated bit. Since we have model inheritance now, we have to use `hasattr` to determine which flavor of `Build` we are using. Then use that in a ternary to determine if we have access to the `branch/hash` or `tag` attributes. If we don't do that, we'll of course get errors on lines 213-215.